lib/service/database.js

```diff
@@ -276,22 +277,37 @@ DatabaseService.prototype.queueAnalysisOperations = function(analysis, callback)
                 }
                 return _status;
             }, Node.STATUS.PENDING);
-            asyncQueries.push(updateNodeAtAnalysisCatalogQuery(nodeToUpdate.id(), status));
+            asyncQueries.push({
+                query: updateNodeAtAnalysisCatalogQuery(nodeToUpdate.id(), status),
+                onerror: updateNodeAtAnalysisCatalogQuery(nodeToUpdate.id(), Node.STATUS.FAILED)
+            });
             nodeToUpdate.setStatus(status);
         });
 
         nodesToQueueForUpdate.forEach(function(nodeToUpdate) {
             if (nodeToUpdate.shouldCacheQuery()) {
                 var targetTableName = nodeToUpdate.getTargetTable();
-                asyncQueries.push(deleteFromCacheTableQuery(targetTableName));
-                asyncQueries.push(updateNodeAtAnalysisCatalogQuery(nodeToUpdate.id(), Node.STATUS.RUNNING));
-                asyncQueries.push(populateCacheTableQuery(targetTableName, nodeToUpdate.sql()));
+                asyncQueries.push({
```

Comment:
What does happen if I send all the queries as objects? Like:

```
var queries = [
  {query: 'select ...'},
  {query: 'select ...', onerror: 'update...'},
  {query: 'select ...',}
]
```

So we don't need `lib/service/job.js`. I think that would also help to keep the inline execution simpler.